examples/download_progress/src/main.rs

```diff
@@ -1,144 +1,277 @@
 use iced::{
-    button, executor, Align, Application, Button, Column, Command, Container,
-    Element, Length, ProgressBar, Settings, Subscription, Text,
+    button, scrollable, window, Align, Application, Button, Color, Column,
+    Command, Container, Element, HorizontalAlignment, Length, ProgressBar, Row,
+    Scrollable, Settings, Space, Subscription, Text,
 };
 
+use crate::download::{Download, Progress};
+
 mod download;
 
-pub fn main() {
-    Example::run(Settings::default())
-}
+const FILE_URL: &str = "https://speed.hetzner.de/100MB.bin";
 
-#[derive(Debug)]
 enum Example {
-    Idle { button: button::State },
-    Downloading { progress: f32 },
-    Finished { button: button::State },
-    Errored { button: button::State },
+    Loading,
+    Loaded(State),
+}
+
+#[derive(Debug, Clone, Default)]
+struct Task {
+    name: String,
+    url: String,
+    button: button::State,
+    state: TaskState,
+}
+
+#[derive(Debug, Clone, Default)]
+struct State {
+    tasks: Vec<Task>,
+    downloading_urls: Vec<String>,
```

Comment:
I think we should be able to remove this list, and simply compute it when necessary by traversing the `tasks` and checking their `state`.